coderd/prometheusmetrics/aggregator.go

```diff
@@ -50,6 +50,8 @@ type MetricsAggregator struct {
 	updateHistogram   prometheus.Histogram
 	cleanupHistogram  prometheus.Histogram
 	aggregateByLabels []string
+	// per-aggregator cache of descriptors
+	descCache map[string]*prometheus.Desc
```

Comment:
I'm just now learning how metrics are collected. To your question about cache strategy, I see there's already some timer based cleanup done by `MetricAggregator`. Might be easy to piggy-back on that for TTL.

I wonder if we can simplify/eliminate the map key and additional eviction by somehow storing a `*prometheus.Desc` alongside the corresponding `annotatedMetric` in the `store` map value. Then I _think_ we could just call `prometheus.NewDesc()` and replace the stored descriptor when needed (which seems like it would be a fairly narrow criteria based on how `hashKey()` uniquely identifies an `annotatedMetric`).